radicle-node/src/tests.rs

```diff
@@ -227,6 +228,91 @@ fn test_inventory_sync() {
     }
 }
 
+#[test]
+fn test_inventory_pruning_stale_entries() {
+    struct Test {
+        limits: Limits,
+        wait_time: LocalDuration,
+        exp_routing_table_size: usize,
+    }
+    let tests = vec![
+        // All zero
+        Test {
+            limits: Limits {
+                routing_max_size: 0,
+                routing_max_age: LocalDuration::from_secs(0),
+            },
+            wait_time: LocalDuration::from_secs(7 * 24 * 60 * 60) + LocalDuration::from_secs(1),
+            exp_routing_table_size: 0,
+        },
+        // All entries are too young to expire.
+        Test {
+            limits: Limits {
+                routing_max_size: 0,
+                routing_max_age: LocalDuration::from_secs(7 * 24 * 60 * 60),
+            },
+            wait_time: LocalDuration::from_secs(7 * 24 * 60 * 60) + LocalDuration::from_secs(1),
+            exp_routing_table_size: 0,
+        },
+        // All entries remain because the table is unconstrained.
+        Test {
+            limits: Limits {
+                routing_max_size: 50,
+                routing_max_age: LocalDuration::from_secs(0),
+            },
+            wait_time: LocalDuration::from_secs(7 * 24 * 60 * 60) + LocalDuration::from_secs(1),
+            exp_routing_table_size: 50,
+        },
+        // Some entries are pruned because the table is constrained.
+        Test {
+            limits: Limits {
+                routing_max_size: 25,
+                routing_max_age: LocalDuration::from_secs(7 * 24 * 60 * 60),
+            },
+            wait_time: LocalDuration::from_secs(7 * 24 * 60 * 60) + LocalDuration::from_secs(1),
+            exp_routing_table_size: 25,
+        },
+    ];
+
+    for test in tests {
+        let mut alice = Peer::config(
+            "alice",
+            Config {
+                limits: test.limits,
+                ..Config::default()
+            },
+            [7, 7, 7, 7],
+            MockStorage::empty(),
+            address::Book::memory().unwrap(),
+            MockSigner::default(),
+            fastrand::Rng::new(),
+        );
+
+        let bob = Peer::new("bob", [8, 8, 8, 8], MockStorage::empty());
+
+        // Tell Alice about the amazing projects available
+        alice.connect_to(&bob);
+        for _peers in 0..5 {
+            alice.receive(
+                &bob.addr(),
+                Message::inventory(
+                    InventoryAnnouncement {
+                        inventory: test::arbitrary::vec::<Id>(10),
+                        timestamp: bob.clock().timestamp(),
+                    },
+                    &MockSigner::default(),
+                ),
+            );
+        }
+
+        // wait for things to happen
+        assert!(test.wait_time > PRUNE_INTERVAL, "pruning must be triggered");
```

Comment:
I'd suggest that we add the PRUNE_INTERVAL to make the above test cases clearer, instead of implying a PRUNE_INTERVAL and the above wait time.

Reply:
The assertion allows us to be exact about testing the 'wait_time'.  Still want to go with sleeping the additional PRUNE_INTERVAL time?